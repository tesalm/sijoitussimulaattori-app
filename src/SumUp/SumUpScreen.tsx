import React from 'react';
import {
  ActivityIndicator,
  ScrollView,
  ToastAndroid,
  TouchableOpacity,
  View,
} from 'react-native';
import { Text } from 'react-native-elements';
import { NavigationScreenProps } from 'react-navigation';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';

import { t } from '../assets/i18n';
import { BidInfo } from '../Bid/reducer';
import { Stock } from '../MarketScreen/reducer';
import { RouteName } from '../navigation/routes';
import { saveTransaction } from '../PortfolioList/actions';
import { SinglePortfolio } from '../PortfolioList/reducer';
import { RootState } from '../redux/reducers';
import { formatCurrency } from '../util/general';
import { sumUpStyles } from './styles';

interface SumUpProps {
  stock?: Stock;
  saveBid: typeof saveTransaction;
  bidInfo: BidInfo;
  portfolio?: SinglePortfolio;
}

type SumUpPropsWithNavigation = SumUpProps & NavigationScreenProps;

interface SumUpState {
  totalCost: string;
  portfolioValueAfter: string;
  dataSent: boolean;
}

export class SumUpScreen extends React.Component<
  SumUpPropsWithNavigation,
  SumUpState
> {
  constructor(props: SumUpPropsWithNavigation) {
    super(props);
    this.state = {
      totalCost: '0',
      portfolioValueAfter: '0',
      dataSent: false,
    };
  }

  static navigationOptions = {
    header: null,
  };

  componentDidMount() {
    if (
      this.props.stock &&
      this.props.stock.stockInfo &&
      this.props.stock.stockInfo.stockMetadata &&
      this.props.portfolio &&
      this.props.portfolio.portfolioInfo.portfolio
    ) {
      const portfolioValue = this.props.portfolio.portfolioInfo.portfolio
        .totalMarketValue;
      this.setState(
        {
          totalCost: formatCurrency(
            this.props.bidInfo.sumOfStocks * this.props.bidInfo.bidLevel,
            this.props.stock.stockInfo.stockMetadata.currency
          ),
        },
        () =>
          this.setState({
            portfolioValueAfter: formatCurrency(
              portfolioValue -
                this.props.bidInfo.sumOfStocks * this.props.bidInfo.bidLevel,
              'USD'
            ),
          })
      );
    }
  }

  componentDidUpdate() {
    if (
      this.props.portfolio &&
      this.props.portfolio.transactionInfo &&
      this.props.portfolio.transactionInfo.transactionSuccess &&
      this.state.dataSent
    ) {
      this.props.navigation.navigate(RouteName.Stock);
      // TODO: Format toast-message for user.
      ToastAndroid.show('Bid was successfull.', ToastAndroid.SHORT);
    }
  }

  countTotalCost() {
    if (
      this.props.stock &&
      this.props.stock.stockInfo &&
      this.props.stock.stockInfo.stockMetadata
    ) {
      return formatCurrency(
        this.props.bidInfo.bidLevel * this.props.bidInfo.sumOfStocks,
        this.props.stock.stockInfo.stockMetadata.currency
      );
    }
    return 0;
  }

  // Saves form-data to backend
  confirmForm() {
    if (this.props.portfolio) {
      const portfolio = this.props.portfolio;
      this.setState({ dataSent: true }, () =>
        this.props.saveBid(
          this.props.bidInfo.action,
          this.props.bidInfo.symbol,
          this.props.bidInfo.sumOfStocks,
          this.props.bidInfo.bidLevel,
          portfolio.uid
        )
      );
    }
  }

  errorMessage() {
    if (this.props.portfolio && this.props.portfolio.transactionInfo) {
      return (
        <Text>
          {String(this.props.portfolio.transactionInfo.transactionsError)}
        </Text>
      );
    }
  }

  render() {
    const { bidInfo, portfolio, stock } = this.props;
    const { totalCost, portfolioValueAfter } = this.state;

    if (
      !stock ||
      !stock.stockInfo ||
      !stock.stockInfo.stockMetadata ||
      !portfolio ||
      !portfolio.portfolioInfo.portfolio
    ) {
      // TODO: Format error for user.
      return <Text>Error!</Text>;
    }

    return (
      <ScrollView style={sumUpStyles.background}>
        <View style={sumUpStyles.headerContainer}>
          <Text style={sumUpStyles.header}>
            {t('SumUpPage.YoureGoingTo')}{' '}
            {<Text style={sumUpStyles.headerHighlight}>{bidInfo.action} </Text>}
            {stock.name}
          </Text>
          <Text style={sumUpStyles.header}>
            {t('SumUpPage.FollowingDetails')}:
          </Text>
        </View>
        <View style={sumUpStyles.detailContainer}>
          <View style={sumUpStyles.detailRowContainer}>
            <View style={sumUpStyles.detailColumnContainer}>
              <Text style={sumUpStyles.valueHeaderSmall}>
                {t('SumUpPage.Portfolio')}
              </Text>
              <Text style={sumUpStyles.valueSmall}>
                {bidInfo.selectedPortfolio}
              </Text>
            </View>
            <View />
            <View style={sumUpStyles.detailColumnContainer}>
              <Text style={sumUpStyles.valueHeaderSmall}>
                {t('SumUpPage.AmountOfStocks')}
              </Text>
              <Text style={sumUpStyles.valueSmall}>{bidInfo.sumOfStocks}</Text>
            </View>
          </View>
          <View style={sumUpStyles.detailRowContainer}>
            <View style={sumUpStyles.detailColumnContainer}>
              <Text style={sumUpStyles.valueHeaderSmall}>
                {t('SumUpPage.TotalValueOfPortfolio')}
              </Text>
              {/* TODO: Replace default currency */}
              <Text style={sumUpStyles.valueSmall}>
                {formatCurrency(
                  portfolio.portfolioInfo.portfolio.totalMarketValue,
                  'USD'
                )}
              </Text>
            </View>
            <View style={sumUpStyles.detailColumnContainer}>
              <Text style={sumUpStyles.valueHeaderSmall}>
                {t('SumUpPage.BidLevel')}
              </Text>
              <Text style={sumUpStyles.valueSmall}>
                {formatCurrency(
                  bidInfo.bidLevel,
                  stock.stockInfo.stockMetadata.currency
                )}
              </Text>
            </View>
          </View>
        </View>
        <View style={sumUpStyles.bottomContainers}>
          <Text style={sumUpStyles.valueHeaderLarge}>
            {t('SumUpPage.TotalCost')}
          </Text>
          <Text style={sumUpStyles.valueLarge}>{totalCost}</Text>
        </View>
        <View style={sumUpStyles.bottomContainers}>
          <Text style={sumUpStyles.valueHeaderLarge}>
            {t('SumUpPage.PortfoliosValueAfter')}
          </Text>
          <Text style={sumUpStyles.valueLarge}>{portfolioValueAfter}</Text>
        </View>

        {portfolio.transactionInfo &&
          portfolio.transactionInfo.transactionsError && (
            <Text style={sumUpStyles.errorMessage}>{this.errorMessage()}</Text>
          )}

        {portfolio.transactionInfo &&
        portfolio.transactionInfo.transactionsLoading ? (
          <View>
            <ActivityIndicator size={'large'} />
          </View>
        ) : (
          <View style={sumUpStyles.confirmCancelButtonContainer}>
            <TouchableOpacity
              style={sumUpStyles.cancelButton}
              onPress={() => this.props.navigation.goBack()}
            >
              <Text style={sumUpStyles.cancelText}>
                {t('SumUpPage.Cancel')}
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              style={sumUpStyles.confirmButton}
              onPress={() => this.confirmForm()}
            >
              <Text style={sumUpStyles.confirmButtonText}>
                {t('SumUpPage.Confirm')}
              </Text>
            </TouchableOpacity>
          </View>
        )}
      </ScrollView>
    );
  }
}

const mapStateToProps = (state: RootState) => ({
  bidInfo: state.bid,
  portfolio: state.portfolioListing.portfolioListing.find((portfolio) => {
    return portfolio.name === state.bid.selectedPortfolio;
  }),
  stock: state.stocksListing.stocks.find((stock) => {
    return stock.symbol === state.stocksListing.currentSymbol;
  }),
});

const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      saveBid: saveTransaction,
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SumUpScreen);
