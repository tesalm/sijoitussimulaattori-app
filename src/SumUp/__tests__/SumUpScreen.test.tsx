import * as React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { SumUpProps, SumUpScreen } from '../SumUpScreen';

describe('sumUp tests', () => {
  const navigationMock: any = {};
  const middlewares: any = [];
  const mockStore = configureStore(middlewares);

  const defaultSumUpProps: SumUpProps = {
    stock: undefined,
    saveBid: jest.fn(),
    bidInfo: {
      action: '',
      bidLevel: 0,
      sumOfStocks: 0,
      selectedPortfolio: '',
      symbol: '',
    },
    portfolio: undefined,
  };

  const sumUpProps: SumUpProps = {
    stock: {
      symbol: 'AAPL',
      name: 'Apple',
      high: 375.53,
      low: 354.38,
      revenue: 3894.34,
      close: 293.42,
      stockInfo: {
        stockMetadata: {
          symbol: 'AAPL',
          name: 'Apple Inc.',
          type: 'string',
          region: 'string',
          marketOpen: '2018-12-12 11:20:00',
          marketClose: '2018-12-12 11:20:00',
          timeZone: 'US/Eastern',
          currency: 'USD',
          fetchTime: new Date('2018-12-17T01:24:00.000Z'),
        },
        intraday: {
          intradayQuote: [
            {
              symbol: 'AAPL',
              date: '2018-12-12T00:20:00.000Z',
              open: 170.4,
              high: 170.43,
              low: 165.43,
              close: 168.63,
              volume: 12279994,
            },
            {
              symbol: 'AAPL',
              date: '2018-12-12T00:10:00.000Z',
              open: 170.4,
              high: 170.43,
              low: 165.43,
              close: 168.63,
              volume: 12279994,
            },
          ],
          fetchTime: new Date('2018-12-17T01:24:00.000Z'),
        },
        historyData: {
          historyDataQuote: [
            {
              symbol: 'AAPL',
              date: '2018-12-15T00:00:00.000Z',
              open: 170.4,
              high: 170.43,
              low: 165.43,
              close: 162.63,
              volume: 12279994,
            },
            {
              symbol: 'AAPL',
              date: '2018-12-14T00:00:00.000Z',
              open: 170.4,
              high: 170.43,
              low: 161.43,
              close: 168.63,
              volume: 12279994,
            },
          ],
          fetchTime: new Date('2018-12-17T01:24:00.000Z'),
        },
        metaLoading: false,
        intraLoading: false,
        historyLoading: false,
        refreshing: false,
        metaError: undefined,
        intraError: undefined,
        historyError: undefined,
      },
    },
    saveBid: jest.fn(),
    bidInfo: {
      action: 'buy',
      bidLevel: 56,
      sumOfStocks: 3,
      selectedPortfolio: 'P1',
      symbol: 'AAPL',
    },
    portfolio: {
      uid: 'Portfolio1',
      name: 'P1',
      balance: 8372,
      ownerId: 'Heidi',
      totalRevenue: 9384,
      totalMarketValue: 484394,
      lastDayRevenue: 483,
      portfolioInfo: {
        loading: false,
        refreshing: false,
        error: undefined,
        portfolio: {
          uid: 'Portfolio1',
          name: 'P1',
          balance: 8372,
          totalRevenue: 9384,
          totalMarketValue: 484394,
          lastDayRevenue: 483,
          ownerId: 'Heidi',
          stocks: [],
        },
      },
      transactionInfo: {
        transactions: undefined,
        transactionsLoading: false,
        transactionsError: undefined,
        transactionSuccess: false,
      },
      transactions: {
        transactionListing: [],
        loading: false,
        error: undefined,
      },
    },
  };

  it('SumUpScreen renders correctly', async () => {
    const component = renderer
      .create(<SumUpScreen {...defaultSumUpProps} {...navigationMock} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('SumUpScreen renders correctly with real data', async () => {
    const component = renderer
      .create(<SumUpScreen {...sumUpProps} {...navigationMock} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
