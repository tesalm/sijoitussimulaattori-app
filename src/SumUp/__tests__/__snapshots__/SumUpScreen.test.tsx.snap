// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sumUp tests SumUpScreen renders correctly 1`] = `
<Text
  style={Object {}}
  theme={
    Object {
      "colors": Object {
        "divider": "#bcbbc1",
        "error": "#ff190c",
        "grey0": "#393e42",
        "grey1": "#43484d",
        "grey2": "#5e6977",
        "grey3": "#86939e",
        "grey4": "#bdc6cf",
        "grey5": "#e1e8ee",
        "greyOutline": "#bbb",
        "primary": "#2089dc",
        "searchBg": "#303337",
        "secondary": "#8F0CE8",
      },
    }
  }
>
  Error!
</Text>
`;

exports[`sumUp tests SumUpScreen renders correctly with real data 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#004D40",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "marginTop": 188,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 40,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
            },
          }
        }
      >
        You're going to
         
        <Text
          style={
            Object {
              "color": "#e8a30c",
              "fontSize": 40,
              "fontWeight": "bold",
            }
          }
          theme={
            Object {
              "colors": Object {
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
              },
            }
          }
        >
          buy
           
        </Text>
        Apple
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 40,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
            },
          }
        }
      >
        with following details
        :
      </Text>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginHorizontal": 32,
          "marginVertical": 32,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "marginHorizontal": 32,
          }
        }
      >
        <View
          style={
            Object {
              "marginBottom": 32,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 24,
              }
            }
            theme={
              Object {
                "colors": Object {
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                },
              }
            }
          >
            Portfolio
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 34,
                "fontWeight": "bold",
              }
            }
            theme={
              Object {
                "colors": Object {
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                },
              }
            }
          >
            P1
          </Text>
        </View>
        <View />
        <View
          style={
            Object {
              "marginBottom": 32,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 24,
              }
            }
            theme={
              Object {
                "colors": Object {
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                },
              }
            }
          >
            Amount of stocks
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 34,
                "fontWeight": "bold",
              }
            }
            theme={
              Object {
                "colors": Object {
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                },
              }
            }
          >
            3
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "marginHorizontal": 32,
          }
        }
      >
        <View
          style={
            Object {
              "marginBottom": 32,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 24,
              }
            }
            theme={
              Object {
                "colors": Object {
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                },
              }
            }
          >
            Total value of portfolio
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 34,
                "fontWeight": "bold",
              }
            }
            theme={
              Object {
                "colors": Object {
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                },
              }
            }
          >
            484394.00 $
          </Text>
        </View>
        <View
          style={
            Object {
              "marginBottom": 32,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 24,
              }
            }
            theme={
              Object {
                "colors": Object {
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                },
              }
            }
          >
            Bid level
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 34,
                "fontWeight": "bold",
              }
            }
            theme={
              Object {
                "colors": Object {
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                },
              }
            }
          >
            56.00 $
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 32,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
            },
          }
        }
      >
        Total cost of transaction:
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 40,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
            },
          }
        }
      >
        168.00 $
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 32,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
            },
          }
        }
      >
        Portfolios value after transaction:
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 40,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
            },
          }
        }
      >
        484226.00 $
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 32,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 72,
            "justifyContent": "center",
            "opacity": 1,
            "width": 224,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 28,
              "fontWeight": "bold",
            }
          }
          theme={
            Object {
              "colors": Object {
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
              },
            }
          }
        >
          CANCEL
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#e8a30c",
            "height": 72,
            "justifyContent": "center",
            "opacity": 1,
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "width": 224,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 28,
              "fontWeight": "bold",
            }
          }
          theme={
            Object {
              "colors": Object {
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
              },
            }
          }
        >
          CONFIRM
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
